### PR TITLE
add Euler Beam

### PR DESCRIPTION
**Pull request by zhiminzhang0830**
### PR types

New features
### PR changes

APIs
### Describe

增加Euler Beam案例

### CODE REVIEW

File: ppsci/equation/pde/euler_beam.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def __init__(self):
+        super().__init__()
+
+        def eulerbeam_compute_func(out):
+            x = out["x"]
+            u = out["u"]
+            eulerbeam = hessian(hessian(u, x), x) + 1
+            return eulerbeam
+
+        self.add_equation("euler_beam", eulerbeam_compute_func)
```
Comment (HydrogenSulfate) [inline ppsci/equation/pde/euler_beam.py:31]: 方程名改为 `Biharmonic`，增加 `dim`、`q`、`D`参数，表示空间维度、载荷、刚度，这样适用1、2、3维的情况。三维公式如下

$$
(\dfrac{\partial^2 }{\partial x} + \dfrac{\partial^2 }{\partial y}+\dfrac{\partial^2 }{\partial z})(\dfrac{\partial^2 }{\partial x} + \dfrac{\partial^2 }{\partial y}+\dfrac{\partial^2 }{\partial z})(u)=\dfrac{q}{D}
$$

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/euler_beam/euler_beam.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+import ppsci
+from ppsci.autodiff import hessian
+from ppsci.autodiff import jacobian
+from ppsci.utils import config
+from ppsci.utils import logger
+
+if __name__ == "__main__":
+    args = config.parse_args()
+    paddle.fluid.core.set_prim_eager_enabled(True)
```
[the remark this answers is not in the document]
Comment (zhiminzhang0830) [reply examples/euler_beam/euler_beam.py:27]: done

File: examples/euler_beam/euler_beam.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    )
+
+    # wrap constraints together
+    constraint = {
+        pde_constraint.name: pde_constraint,
+        bc.name: bc,
+    }
+
+    # set optimizer
+    optimizer = ppsci.optimizer.Adam(learning_rate=0.001)([model])
```
[the remark this answers is not in the document]
Comment (zhiminzhang0830) [reply examples/euler_beam/euler_beam.py]: done

File: examples/euler_beam/euler_beam.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    # directly evaluate model from pretrained_model_path(optional)
+    logger.init_logger("ppsci", f"{output_dir}/eval.log", "info")
+    solver = ppsci.solver.Solver(
+        model,
+        constraint,
+        output_dir,
+        equation=equation,
+        validator=validator,
+        visualizer=visualizer,
+        pretrained_model_path=f"./{output_dir}/checkpoints/best_model",
```
[the remark this answers is not in the document]
Comment (zhiminzhang0830) [reply examples/euler_beam/euler_beam.py]: dne

File: ppsci/equation/pde/euler_beam.py (hunk 1 above)
Comment (zhiminzhang0830) [reply ppsci/equation/pde/euler_beam.py]: 已修改

File: examples/euler_beam/euler_beam.py [hunk 5]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import paddle
+from regex import D
```
Comment (HydrogenSulfate) [inline examples/euler_beam/euler_beam.py:16]: 删除无用 import

File: examples/euler_beam/euler_beam.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            "u__x__x__x": lambda d: jacobian(hessian(d["u"], d["x"]), d["x"])[3:4],
+        },
+        {"u0": 0, "u__x": 0, "u__x__x": 0, "u__x__x__x": 0},
+        geom["interval"],
+        {**dataloader_cfg, "batch_size": 4},
+        ppsci.loss.MSELoss("sum"),
+        evenly=True,
+        name="BC",
+    )
+
```
Comment (HydrogenSulfate) [inline examples/euler_beam/euler_beam.py:76]: 76行的空行删除

File: examples/euler_beam/euler_beam.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        bc.name: bc,
+    }
+
+    # set optimizer
+    optimizer = ppsci.optimizer.Adam(learning_rate=0.001)((model,))
+
+    def u_solution_func(out):
+        """compute ground truth for u as label data"""
+        x = out["x"]
+        return -(x**4) / 24 + x**3 / 6 - x**2 / 4
```
Comment (HydrogenSulfate) [inline examples/euler_beam/euler_beam.py:89]: 该函数移动至set validator下方

File: ppsci/equation/pde/biharmonic.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            u = out["u"]
+
+            keys = ["x", "y", "z"]
+            biharmonic = -self.q / self.D
+            for i in range(self.dim):
+                key = keys[i]
+                u__ = hessian(u, out[key])
+                for j in range(self.dim):
+                    key = keys[j]
+                    biharmonic += hessian(u__, out[key])
```
Comment (HydrogenSulfate) [inline ppsci/equation/pde/biharmonic.py:44]:
``` py
invars = ("x", "y", "z")[:self.dim]
for invar_i in invars:
    for invar_j in invars:
        biharmonic += hessian(hessian(u, invar_i), invar_j)
```

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/euler_beam/euler_beam.py (hunk 6 above)
Comment (zhiminzhang0830) [reply examples/euler_beam/euler_beam.py]: done

File: examples/euler_beam/euler_beam.py (hunk 7 above)
Comment (zhiminzhang0830) [reply examples/euler_beam/euler_beam.py:90]: done

File: ppsci/equation/pde/biharmonic.py (hunk 8 above)
Comment (zhiminzhang0830) [reply ppsci/equation/pde/biharmonic.py]: eone

**Review by HydrogenSulfate — APPROVED** [review]
